akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -62,11 +109,7 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with GenericOut
           if (endStream) (Source.empty, HalfClosedRemote)
           else {
             val subSource = new SubSourceOutlet[ByteString](s"substream-out-$streamId")
-            (Source.fromGraph(subSource.source), Open(new BufferedOutlet[ByteString](subSource) {
-              override def onDownstreamFinish(): Unit =
-                multiplexer.pushControlFrame(RstStreamFrame(streamId, ErrorCode.CANCEL))
-              incomingStreams -= streamId
```

Comment:
Yes, without that statement in the `onDownstreamFinish` we would have leaked memory for already closed streams.